### ISSUE TITLE
Ability to provide default value when using t func like this i18n.t('namespace:title', "Default value")

### ISSUE THREAD

**Issue by saravanan10393** [i18next/i18next#1085]
I am generating resource json in my build step using `i18next.scanner`. `Trans`  component generates json like `{login:title: <0>Title </0>}`.  But when using `i18n.t('title')` it gives json with only keys, not values, like this `{login:title: ""}`. My aim is to automate translation process by sending this generated json file to some server and get the translation done in the build process. Currently I am struck with passing default value to `t func`. 

I am in need of this requirement to move forward. any help. 

using code like below: 

`
  i18n.t('login:title', "Welcome")
`

wil result in login.json
`
 {
  "title": "Welcome"
 }
`

**Comment by jamuhl** [comment]
What is the problem? scanner not extracting the defaultValue of "Welcome"? As `i18n.t('login:title', "Welcome")` is the correct way to pass default to t

**Comment by saravanan10393** [comment]
Yes @jamuhl , `i18n.t('login:title', "Welcome")` if it is right way to do it, then problem with Scanner. Scanner is not extracting the value of Welcome.

**Comment by jamuhl** [comment]
Eventual extraction of `i18n.t('login:title', { defaultValue: "Welcome" })` works -> might be worth a try.

Alternative is not using extraction at all but continuous localization tooling like our https://locize.com -> turning on saveMissing in i18next will automatically add new key and defaultValues to your localization project (only downside - does not come for free)

**Comment by saravanan10393** [comment]
Thanks @jamuhl, I thought to use [https://locize.com ](https://locize.com) first step is to extract all the used messages, I am wondering that this process will happens automatically by just adding script from [https://locize.com ](https://locize.com)

**Comment by jamuhl** [comment]
You replace the xhr backend with https://github.com/locize/i18next-locize-backend (or in case of node.js with the node.js backend)...add your project and apiKey -> set saveMissing: true on i18next init. Every call to i18next.t which results in no translation found (missing) will send that key and defaultValue to locize on the fly.

saveMissing works also with the regular xhr backend - but you will need a backend handling the request for adding keys (eg. using the i18next express middleware)
